common/src/main/java/net/whimxiqal/journey/data/DataVersion.java

```diff
@@ -86,51 +91,124 @@ static DataVersion version() {
     }
   }
 
   static DataVersion readVersion() {
     File versionFile = Journey.get().proxy().dataFolder().resolve(VERSION_FILE_NAME).toFile();
-    Scanner scanner;
-    try {
-      scanner = new Scanner(versionFile);
-    } catch (FileNotFoundException e) {
-      e.printStackTrace();
-      return DataVersion.ERROR;
-    }
-    if (!scanner.hasNextLine()) {
-      Journey.logger().error("The database version file is empty. Please delete your database files and restart.");
-      return DataVersion.ERROR;
+
+    if (versionFile.exists()) {
+      versionFile.delete();
+      writeVersion(DataVersion.V001);
+      return DataVersion.V001;
     }
-    DataVersion version = DataVersion.fromString(scanner.nextLine());
-    if (version == DataVersion.ERROR) {
-      Journey.logger().error("Journey's database is in an error state. Please delete your database files and restart.");
+
+    switch (Settings.STORAGE_TYPE.getValue()) {
+      case MYSQL -> {
+        MySqlConnectionController mysqlController = new MySqlConnectionController();
+        try (Connection connection = mysqlController.establishConnection()) {
+          PreparedStatement statement = connection.prepareStatement("SHOW TABLES LIKE 'journey_db_version'");
+          ResultSet resultSet = statement.executeQuery();
+
+          if (!resultSet.next()) {
+            PreparedStatement createTableStatement = connection.prepareStatement("CREATE TABLE journey_db_version (db_version int);");
+            createTableStatement.execute();
+          }
+
+          PreparedStatement readStatement = connection.prepareStatement("SELECT Max(db_version) as DBVersion FROM journey_db_version;");
+          ResultSet readResultSet = readStatement.executeQuery();
+
+          if (readResultSet.next()) {
+            return DataVersion.fromString(String.valueOf(readResultSet.getInt("DBVersion")));
+          } else return DataVersion.V000;
+        } catch (SQLException e) {
+          e.printStackTrace();
+        }
+      }
+      case SQLITE -> {
+        String sqliteAddress = "jdbc:sqlite:" + Journey.get().proxy().dataFolder() + "/" + DATABASE_FILE_NAME;
+        SqliteConnectionController sqliteController = new SqliteConnectionController(sqliteAddress);
+
+        try (Connection connection = sqliteController.establishConnection()) {
+          PreparedStatement statement = connection.prepareStatement("SELECT name FROM sqlite_master WHERE type='table' AND name='journey_db_version'");
+
+          ResultSet resultSet = statement.executeQuery();
+
+          if (!resultSet.next()) {
+            PreparedStatement createTableStatement = connection.prepareStatement("CREATE TABLE journey_db_version (db_version int);");
+            createTableStatement.execute();
+          }
+
+          PreparedStatement readStatement = connection.prepareStatement("SELECT Max(db_version) as DBVersion FROM journey_db_version;");
+          ResultSet readResultSet = readStatement.executeQuery();
+
```

Comment:
There should be an index on `db_version`

Reply:
Added the index.